### PR TITLE
[Fix #5897] Apply symbol and word array conversion to array of size 1

### LINKED ISSUE #5897: Style/SymbolArray ignores arrays of 1 element, even when MinSize is set to 0 (With root cause)
## Expected behavior

When using:

```
Style/SymbolArray:
  Enabled: true
  MinSize: 1
```

I expect `[:symbol]` to be auto-corrected to `%i[symbol]`. 

## Actual behavior

Rubocop ignores `[:symbol]`. 

The issue is caused by [this line](https://github.com/bbatsov/rubocop/blob/3988d2ffb28e57f1f5b6075762d0feee422bb513/lib/rubocop/cop/mixin/array_syntax.rb#L11), which does a check of `node.values.size > 1` when it should be doing a check of `node.values.size >= 1`.

## Steps to reproduce the problem

1. Create a `file.rb` with an array of one symbols `[:symbol]`
2. Enable `Style/SymbolArray` with a `MinSize` of 1 or 0. 
3. Run `rubocop --auto-correct file.rb`

The array is not detected as an issue, and is not corrected.

## RuboCop version

```
└> rubocop -V
0.55.0 (using Parser 2.5.1.0, running on ruby 2.5.1 x86_64-linux)
```

### PR DESCRIPTION
**Pull request by TikiTDO**
Ensures Style/SymbolArray and Style/WordArray work on arrays of size 1 when MinSize allows it.

Fixes #5897

### CODE REVIEW

**Comment by TikiTDO** [comment]
Because previously these rules ignores arrays of size one, despite the fact that it was set to default `MinSize` of `0` a lot of warnings did not exist. In order to keep consistent behavior I could update the default `MinSize` of SybmolArray and WordArray to 2.

File: lib/rubocop/cop/mixin/array_syntax.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ module ArraySyntax
       private
 
       def bracketed_array_of?(element_type, node)
-        return false unless node.square_brackets? && node.values.size > 1
+        return false unless node.square_brackets? && node.values.size >= 1
```
[the remark this answers is not in the document]
Comment (TikiTDO) [reply lib/rubocop/cop/mixin/array_syntax.rb]: Good call

[the remark this answers is not in the document]
Comment (TikiTDO) [reply lib/rubocop/cop/mixin/array_syntax.rb]: Though at that point maybe it would be best to just do

`return false if node.values.empty? || !node.square_brackets?`

I always find `unless` starts to give me pause when the condition has more than one expression

[the remark this answers is not in the document]
Comment (TikiTDO) [reply lib/rubocop/cop/mixin/array_syntax.rb]: Actually the one you suggested read the easiest of the various variations I tried.

**Comment by bbatsov** [comment]
> Because previously these rules ignores arrays of size one, despite the fact that it was set to default MinSize of 0 a lot of warnings did not exist. In order to keep consistent behavior I could update the default MinSize of SybmolArray and WordArray to 2.

I think MinSize 0 meant that the size doesn't matter and we should keep this. You fix a bug, the current behaviour was a mistake, not something we have to keep.

**Comment by TikiTDO** [comment]
@bbatsov At first I tried not changing MinSize, but when the spec ran the rubocop self-test it started trying to complain about things like the spec file to percent-arrays ([See here](https://travis-ci.org/bbatsov/rubocop/builds/380903073?utm_source=github_status&utm_medium=notification)). 

I figured this could cause confusion if I pushed it through as it was, though I erred on the side of keeping the behavior consistent.

**Comment by bbatsov** [comment]
Hmm, I'll have to think about this. A value like 2 seems completely random to me. And it seems that this should have been better documented in the cop to begin with.

**Comment by deivid-rodriguez** [comment]
@bbatsov Note though that the value of "2", which one might argue is random indeed, was previously hardcoded and configurations with values below it ignored. So this bug fix seems independent from the discussion of a potential change in the default.

The value of "2" I'd guess comes from the fact that in the case of single element arrays there's a single extra character using the standard notation, so the `%` syntax doesn't save much typing. I'm unsure whether that merits a special case or not... :thinking:

**Comment by bbatsov** [comment]
I get this, but it still feel arbitrary. Anyways, I'm with accepting this fix as is and figuring out the rest down the road.

**Comment by bbatsov** [comment]
Hmm, just noticed the PR is missing a changelog entry.

**Comment by deivid-rodriguez** [comment]
I agree, it feels arbitrary.... But using the notation on single arrays feels a bit weird.. But maybe it just feels weird because rubocop has had this default for a long time... :man_shrugging:

**Comment by TikiTDO** [comment]
Whoops, my bad. Added changelog entry

File: spec/rubocop/cop/style/word_array_spec.rb [hunk 2]
```diff
@@ -82,10 +82,6 @@
       expect_no_offenses('%w(one two three)')
     end
 
-    it 'does not register an offense for array with one element' do
-      expect_no_offenses('["three"]')
-    end
-
```
[the remark this answers is not in the document]
Comment (TikiTDO) [reply spec/rubocop/cop/style/word_array_spec.rb]: It feels a weird to include that in the `word_array` and `symbol_array` cops, since that's really a test of the `array_syntax.rb` mixin. There's also already a test that sort of coverts the non-zero `MinSize`, though it may be better to just be more explicit about it.

Thoughts?

```ruby
  it 'detects right value for MinSize to use for --auto-gen-config' do
      inspect_source(<<-RUBY.strip_indent)
        [:one, :two, :three]
        %i(a b c d)
      RUBY

      expect(cop.offenses.size).to eq(1)
      expect(cop.messages).to eq(['Use `%i` or `%I` for an array of symbols.'])
      expect(cop.config_to_allow_offenses).to eq('EnforcedStyle' => 'percent',
                                                 'MinSize' => 4)
    end
```

[the remark this answers is not in the document]
Comment (TikiTDO) [reply spec/rubocop/cop/style/word_array_spec.rb]: I added two specs to explicitly test the MinSize > 1 behavior

**Comment by bbatsov** [comment]
👍
